aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -633,7 +633,9 @@ def _call_route(self, route: Route, args: Dict[str, str]) -> ResponseBuilder:
 
             raise
 
-    def not_found(self, func: Callable):
+    def not_found(self, func: Optional[Callable] = None):
+        if func is None:
```

Comment:
for future note, if this method ever supports keyword arguments, `func` will also be None and will require a change to L638

example: https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/aws_lambda_powertools/tracing/tracer.py#L286